### DISCUSSION TITLE
My 2019 Model 3 is still working on the old API...is this normal?

### DISCUSSION THREAD

**Discussion by mattenz** [alandtse/tesla#931] [General]
Everything I've read seems to indicate that my car should no longer accept commands using the old API yet it's still currently working with this integration on 3.20.4 without going through the process to migrate to the new fleet API. I'm in New Zealand if that makes a difference.

I'm guessing it's no point switching over while the old API is still working with my car?

**Comment by skipishere** [comment]
Yes it's normal-ish, my 2023 Model 3 still happily works on the old API.

I have started the ground work to use the new way just in case it goes off soon

**Reply by mattenz** [reply]
I just noticed that Tesla has very recently updated [this page](https://developer.tesla.com/docs/fleet-api#2024-03-26-shutting-down-legacy-vehicle-api-endpoints) to state that April is now the shutdown date. It also states

> Per the Fleet API policies, continued usage of unsupported APIs will result in the revocation of Fleet API access privileges.

which reads to me like we’ll be punished if we don’t switch?

**Reply by skipishere** [reply]
If they did anymore than "it doesn't work with the old API" then I'd be worried, if they actively then blocked your car from working with any new API going forward that would be ridiculous and very bad behaviour on their part.

**Comment by robupham** [comment]
My 2020 (pre-facelift) M3 running 2024.8.7 went offline this morning (I'm in the UK). I've setup the HTTP Proxy succesfully (with dev credentials, etc.) but still can't get the Tesla integration to kick back into life. It claims it's reconfigured correctly but still fails.

I'd be interest to know if any pre-facelift Model 3 owners have this new setup working.
